src/datadog/trace_segment.cpp

```diff
@@ -78,6 +83,22 @@ void inject_trace_tags(
   }
 }
 
+Expected<SamplingDecision> parse_sampling_delegation_response(
+    StringView response) {
+  try {
+    auto json = nlohmann::json::parse(response);
+
+    SamplingDecision sampling_decision;
+    sampling_decision.origin = SamplingDecision::Origin::DELEGATED;
+    sampling_decision.priority = json["priority"];
+    sampling_decision.mechanism = json["mechanism"];
+
+    return sampling_decision;
+  } catch (...) {
+    return Error{Error::Code::OTHER, "Could not parse"};
```

Comment:
Dedicated error codes would be good here (how many cases you want to individually cover is up to you), and make sure that the error message includes whatever nlohmann complained about (`exception.what()`...).

Reply:
Resolved in 4c20da00573f307d44b3554528baf3c22727da06